include/openthread/platform/border_routing.h

```diff
@@ -65,6 +65,18 @@ extern "C" {
  */
 extern void otPlatBorderRoutingProcessIcmp6Ra(otInstance *aInstance, const uint8_t *aMessage, uint16_t aLength);
 
+/**
+ * Used to notify the platforms whether they should request an prefix via DHCPv6 PD for Thread interface.
+ * When aShouldRequest is false, the platform must release the prefix lease.
+ *
+ * Requires `OPENTHREAD_CONFIG_BORDER_ROUTING_DHCP6_PD_ENABLE`.
+ *
+ * @param[in] aInstance  A pointer to an OpenThread instance.
+ * @param[in] aRequest  Indicate if the platform need to request/release the prefix.
+ *
+ */
+void otPlatBorderRoutingHandleDhcp6PdStateChange(otInstance *aInstance, bool aRequest);
```

Comment:
It may be better to frame platform API name as **a command that directly instruct the platform to perform specific actions**.  It is up to platform layer to decide how to implement this but it should be clear what we are asking platform to do.

- In this case, we are asking the platform to request or release DHCP6 Prefix
   - I suggest `otPlatBorderRoutingRequestDhcp6Prefix()` and `otPlatBorderRoutingReleaseDhcp6Prefix()` to be clear about the desired actions.
- Or we can alternatively think that we are enabling/disabling DHCP6 PD at the platform layer:
   - So can use `otPlatBorderRoutingDhcp6PdSetEnabled(otInstance *, bool aEnable)`to convey this.

The name `otPlatBorderRoutingHandleDhcp6StateChange()` sound as if we are notifying platform instead of  controlling/instructing it.

----

> For querying the state, the app should call `otBorderRoutingDhcp6PdGetState`.

> I would prefer passing the state enum instead of a boolean to the platform.

We have layered architecture:
- OpenThread (OT) Public APIs: These APIs provide control over the OT stack's behavior and allow higher-layer code to initiate specific functions/actions.
- OT Platform APIs: The OT stack utilizes these APIs to interact with and instruct the underlying platform layer.

It is always good to have layer separation and avoid cross-layer interactions. 

- The `otBorderRoutingDhcp6PdGetState()` API (and `otBorderRoutingDhcp6PdState` enum) are part of public OT APIs, and therefore primarily intended for consumption by applications and higher-layer components. 
- The platform layer implementation of Prefix Delegation (PD) should typically obtain the necessary  information directly from its internal data and/or from `otPlat` API calls and not rely or use any public OT APIs.

Reply:
`OnDhcp6PdStateChange()` brings the question of "which state" and is notifying platform rather than instructing it to take an action. 

This name reflects how the OT stack intends to use this API (i.e., whenever its internal PD prefix state gets changed) rather than what command/instruction is being asked of the platform layer to perform. Platform should not care how/what states are being tracked by OT stack (related to Prefix PD feature) and how the API will be used.

Also would be good to adhere to common OT API naming convention of "verb + [noun(s)]" to help improve consistency and readability. Example:`otPlatRadioTransmit`, `otPlatSettingsAdd`, `otPlatInfraIfSendIcmp6Nd()`, `otPlatBorderRoutingProcessIcmp6Ra()`, `otPlatDsoEnableListening()`.

Reply:
otPlatBorderRoutingRequestDhcp6Pd()

Reply:
👍